modules/microshift-cni.adoc

```diff
@@ -0,0 +1,301 @@
+
+//Q: this module is not done; included for the sake of the build
+//[IMPORTANT]
+//====
+//The default CNI configuration is intended to match the developer environment described in MicroShift Development Environment on RHEL 8.
+//====
+//Q: What's the default production configuration?
+
+{product-title} uses OVN-Kubernetes, a virtualized network for pods and services. The OVN-Kubernetes Container Network Interface (CNI) plug-in is the network provider for the cluster. OVN-Kubernetes is based on Open Virtual Network (OVN) and provides an overlay-based networking implementation.
+
+A cluster that uses the OVN-Kubernetes network provider also runs Open vSwitch (OVS) on the node. OVN configures OVS on the node to implement the declared network configuration. OVS runs as a systemd service on the {product-title} node. The OVS RPM package is installed as a dependency to the `microshift-networking` RPM package. With {product-title}, each deployment requires configuration. OVS is started immediately when the `microshift-networking` RPM is installed.
+//Q: can customers copy network settings from a first node onto others, or do them all at once?
+//Q: is this conceptually correct = one microshift install can create several single-node deployments? or is it one-to-one and a new microshift instance has to be installed for every single node deployment?
+
+By default, three performance optimizations are applied to OVS services to minimize resource consumption:
+
+* CPU affinity to `ovs-vswitchd.service` and `ovsdb-server.service`
+* `no-mlockall` to `openvswitch.service`
+* Limit handler and `revalidator` threads to `ovs-vswitchd.service`
+
+== Network features
+A wide range of networking features are available with OVN-Kubernetes, such as:
+
+* Kubernetes network policy
+* Dynamic node IP
+* Cluster network on specified host interface
+* Secondary gateway interface
+* Dual stack
+* Service idling (disabled)
+* Egress firewall/QOS/IP (disabled)
+* Cluster network on specified host interface
+//these don't match the default values in the OCP docs; what features are available/relevant to MicroShift
+//https://docs.openshift.com/container-platform/4.11/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html#nw-ovn-kubernetes-matrix_about-ovn-kubernetes
+
+== {product-title} networking components overview
+
+This brief overview describes networking components and their operation in {product-title}.
+//Q: are there immutable settings we should tell users about?
+NetworkManager::
+NetworkManager is required by OVN-Kubernetes to set up the initial gateway bridge on the {product-title} node. NetworkManager and NetworkManager-OVS RPM packages are installed as dependencies to the `microshift-networking` RPM package. NetworkManager is configured to use the `keyfile` plugin by default and restarts after installation of the `microshift-networking` RPM package.
+//Q: Meaning the microshift networking RPM replaces the keyfile plugin?
```

Comment:
I don't know what's the keyfile plugind 

microshift-networking is a package that pulls in any networking related dependencies and some systemd services to initialize networking (like microshift-ovs-init)

Reply:
@ggiguash Do you know the answer to this Q about the keyfile plugin? Is this relevant for the production environment end user?

Reply:
A keyfile is one way of configuring NetworkManager and specifies a specific type of configuration file to use. https://developer-old.gnome.org/NetworkManager/stable/nm-settings-keyfile.html

The plugin is not replaced. We drop the config files in place then restart the service to ensure they're picked up and read.

Reply:
yep, there are two plugins supported by NM, ifcfg-rh and keyfile. In microshift, we force to use keyfile plugin by updating the NM systemd config. It is not to replace, but switch the plugin config.